File: challenges/Python/Easy/استبدال The.py
def replaceThe(txt: str) -> str:
    words = txt.split()

    for i, word in enumerate(words):
        if word == "the":
            if words[i+1][0] in ['a', 'e', 'i', 'o', 'u']:
                words[i] = "an"
            else:
                words[i] = "a"

    return " ".join(words)

File: challenges/JavaScript/Easy/استبدال The.js
/*

Creator: OmarAfet
https://profile.satr.codes/OmarAfet/public/overview
https://github.com/OmarAfet

*/

function replaceThe(txt) {
	let arr = txt.split(" ");
	for (let i = 0; i < arr.length; i++) {
		if (arr[i] === "the" && arr[i + 1][0].match(/[aeiou]/)) {
			arr[i] = "an";
		} else if (arr[i] === "the") {
			arr[i] = "a";
		}
	}
	return arr.join(" ");
}
